### PR TITLE
BUFG Support

### PR DESCRIPTION
**Pull request by reillymck**
Add bufg to phys netlist.
Example BUFG -> BUFGCTRL:
```
BUFGCTRL #(
    .INIT_OUT(0),
    .IS_CE0_INVERTED(1'b0),
    .IS_CE1_INVERTED(1'b1),
    .IS_IGNORE0_INVERTED(1'b1),
    .IS_IGNORE1_INVERTED(1'b0),
    .IS_S0_INVERTED(1'b0),
    .IS_S1_INVERTED(1'b1),
    .PRESELECT_I0("TRUE"),
    .PRESELECT_I1("FALSE")) 
    clk_IBUF_BUFG_inst_phys
       (.CE0(\<const1> ),
        .CE1(\<const1> ),
        .I0(clk_IBUF),
        .I1(\<const1> ),
        .IGNORE0(\<const1> ),
        .IGNORE1(\<const1> ),
        .O(clk_IBUF_BUFG),
        .S0(\<const1> ),
        .S1(\<const1> ));
```

fasm2bels example:
```
BUFGCTRL #(
    .INIT_OUT(0),
    .IS_CE0_INVERTED(0),
    .IS_CE1_INVERTED(1),
    .IS_IGNORE0_INVERTED(1),
    .IS_IGNORE1_INVERTED(0),
    .IS_S0_INVERTED(0),
    .IS_S1_INVERTED(1),
    .PRESELECT_I0("TRUE"),
    .PRESELECT_I1("FALSE")
  ) CLK_BUFG_BOT_R_X139Y152_BUFGCTRL_X0Y0_BUFGCTRL (
.CE0(1'b1),
.CE1(1'b1),
.I0(LIOB33_X0Y127_IOB_X0Y128_I),
.I1(1'b1),
.IGNORE0(1'b1),
.IGNORE1(1'b1),
.O(CLK_BUFG_BOT_R_X139Y152_BUFGCTRL_X0Y0_O),
.S0(1'b1),
.S1(1'b1)
  );
```

### CODE REVIEW

**Review by jgoeders — APPROVED** [review]

File: bfasst/transform/xilinx_phys_netlist.py [hunk 1]
```diff
@@ -139,6 +141,11 @@ def run_rapidwright(self, design, phys_netlist_checkpoint, phys_netlist_edif_pat
         # Get the LUT6_2 EDIFCell (all LUTs will be replaced with equivalent LUT6_2 primitives)
         lut6_2_edif_cell = netlist.getHDIPrimitive(Unisim.LUT6_2)
 
+        # Init BUFGCTRL cell template
+        self.bufgctrl_edif_cell = netlist.getHDIPrimitive(Unisim.BUFGCTRL)
+        self.const1_net = rw_design.getNetlist().getNetFromHierName("<const1>")
+        assert self.const1_net
```
[the remark this answers is not in the document]
Comment (jgoeders) [reply bfasst/transform/xilinx_phys_netlist.py:147]: As I mentioned, I rather not hard-code <const1>.  Can you get the Unisim.VDD cell, perhaps search for it in the design, and add it if it doesn't already exist?

[the remark this answers is not in the document]
Comment (jgoeders) [reply bfasst/transform/xilinx_phys_netlist.py:147]: Hmm, that's interesting.  I'm not sure why it's happening like that.  I guess just add the VCC instance and we can see if that works and go from there.

**Comment by reillymck** [comment]
Right now, EDIF parsing fails when vivado tries to load the new design after the physical transformation python scripts. I add a new VCC cell, but for some reason it does not appear right. (I know I am missing something). Currently, the cell appears like so in the new EDIF file. It appears in the library hdi_primitives section instead of the work library. So Vivado complains that it does not know what clk_IBUF_BUFG_inst_phys is.
```
(cell VCC (celltype GENERIC)
     (view netlist (viewtype NETLIST)
       (interface 
        (port P (direction OUTPUT))
       )
       (contents
         (net vcc_net_phys_netlist (joined
          (portref CE0 (instanceref clk_IBUF_BUFG_inst_phys))
          (portref CE1 (instanceref clk_IBUF_BUFG_inst_phys))
          (portref I1 (instanceref clk_IBUF_BUFG_inst_phys))
          (portref IGNORE0 (instanceref clk_IBUF_BUFG_inst_phys))
          (portref IGNORE1 (instanceref clk_IBUF_BUFG_inst_phys))
          (portref S0 (instanceref clk_IBUF_BUFG_inst_phys))
          (portref S1 (instanceref clk_IBUF_BUFG_inst_phys))
          (portref P (instanceref vcc_phys_netlist))
          )
         )
       )
     )
   )
```
There is a VCC instance in the work library:
```
(instance VCC (viewref netlist (c
[Sout_reg_impl_physical_edf.txt](https://github.com/byuccl/bfasst/files/11066689/Sout_reg_impl_physical_edf.txt)
ellref VCC (libraryref hdi_primitives))))
```
I am not sure how to access the work library instance. I am still debugging this, but I thought I would see if you already know what I am doing wrong. I have attached the full edif output file post-physical transformation.
[Sout_reg_impl_physical_edf.txt](https://github.com/byuccl/bfasst/files/11066690/Sout_reg_impl_physical_edf.txt)

**Comment by jgoeders** [comment]
https://github.com/byuccl/bfasst/pull/140/commits/397af9d9ea589915bb714eedaf79e0973e0b4ea7

The issue was with how you were creating the inst/net.  They need to be created as part of the top-level cell in the design.

**Pull request merged**